packages/database/lib/DatabaseSyncTree.js

```diff
@@ -263,7 +263,8 @@ class DatabaseSyncTree {
 
     if (once) {
       const subscription = SharedEventEmitter.addListener(eventRegistrationKey, event => {
-        this._onOnceRemoveRegistration(eventRegistrationKey, listener, event);
+        const wrappedListener = this._onOnceRemoveRegistration(eventRegistrationKey, listener, event);
+        wrappedListener(event);
```

Comment:
not entirely sure if i am missing something here 😆

Reply:
@mikehardy that method returns the function...but where is the function called?

Reply:
this line *right here* calls it https://github.com/invertase/react-native-firebase/blame/b8f80a41e864c2c8d97602df625abdd2ee6bc347/packages/database/lib/DatabaseSyncTree.js#L323

regarding skipped test, it feels like we are mixing things here. Are we talking about once, or on? You are pointing to _onOnceRemoveRegistration, but you are talking about an 'on' failure. I can't put those together.

Reply:
Sorry, maybe I am getting confused, but let me try to break this down a bit better.

1. https://github.com/invertase/react-native-firebase/blame/b8f80a41e864c2c8d97602df625abdd2ee6bc347/packages/database/lib/DatabaseSyncTree.js#L323
This line does indeed called the passed listener, but _onOnceRemoveRegistration is returning a function (that calls the passed listener), but that function does not appear to be called

> regarding skipped test, it feels like we are mixing things here. Are we talking about once, or on? You are pointing to _onOnceRemoveRegistration, but you are talking about an 'on' failure.

2. To speak to this a bit better -
Let's take a look at this snippet: 
https://github.com/invertase/react-native-firebase/blob/master/packages/database/lib/DatabaseQuery.js#L313

This is the ref.on method. Notice, it passes `once` as true when registering the cancellation callback. I am assuming `once` is passed as true, because we unsubscribe after triggering the error event / cancel callback


Now if we look at `addRegistration`, 
https://github.com/invertase/react-native-firebase/blame/b8f80a41e864c2c8d97602df625abdd2ee6bc347/packages/database/lib/DatabaseSyncTree.js#L264

It checks to see if `once` is true, and then 
```
      const subscription = SharedEventEmitter.addListener(eventRegistrationKey, event => {
        this._onOnceRemoveRegistration(eventRegistrationKey, listener, event);
        subscription.remove();
      });
```
I hope that clears up 
```
e. Are we talking about once, or on? You are pointing to _onOnceRemoveRegistration, but you are talking about an 'on' failure.
```

Reply:
I don't have time to look at this more deeply right now, sorry, I'll queue it for later but I didn't see anything wrong in the code each of the times I've responded and looked again.